HelpSource/Classes/NotificationCenter.schelp

```diff
@@ -4,26 +4,60 @@ related:: Classes/SimpleController, Classes/NodeWatcher
 categories:: Control
 
 description::
-One common OOP pattern is Model-View-Controller where one object (the controller) is a dependant of the model. Every time the model changes it notifies all of its dependants. In this case the model has a dictionary of dependants and iterates through those.
 
-Another common pattern is NotificationCenter wherein an object emits a notification and clients can register functions that will be executed when that notification happens.
+In certain cases it may be useful to make an object perform a specific function when receiving a predefined message from another given object.  For example, a GUI element may want to update its state when the state of a server object changes.  Such cases are: Update the count of synthdefs when a new synthdef is loaded, update the enabled/disabled state of a control gui element when a Synth which it controls starts running or is freed, etc. NotificationCenter makes it possible to add such a custom function to any object called the observer, and to perform it whenever another object called the subject issues a notification identified by a symbol called the message.
 
-A link::Classes/Server:: emits a \newAllocators notification when it creates new node and bus allocators which it does when it quits or boots.
+The implementation provided by NotificationPattern is a variant of a common OOP pattern called the Observer Pattern, which provides a way for an object to flexibly broadcast messages to interested receiver objects.  This pattern defines a one-to-many dependency between objects so that when one object changes state, all its dependents are notified and updated automatically.  (see: https://www.gofpatterns.com/behavioral-design-patterns/behavioral-patterns/observer-pattern.php and https://en.wikipedia.org/wiki/Observer_pattern ).  An object that  notifies its dependents is called the subject, and the dependents are called observers.
+
+The original implementation of this pattern in SuperCollider involves three steps:
+
+NUMBEREDLIST::
+## A subject adds a dependant with: "subject.addDependant(observer)"
+## When a subject wants to inform its observers (dependants) that it has changed, it uses the method changed: "subject.changed(<optional arguments>...)"
+## The method changed obtains the list of dependants of the subject and iterates over each observer sending it the message "update": "observer.changed(subject, <optional arguments>...)"
```

Comment:
you can use `code:: receiver.message(a, b, c) ::` for inline code, please change these to use that